toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -334,3 +335,44 @@ func unbuiltPackagesCSVRows(pkgGraph *pkggraph.PkgGraph, unbuiltPackages, failed
 
 	return
 }
+
+// Helper function to print summarized numbers of the build to the logger.
+func printSummary(failedSRPMs, failedSRPMsTests map[string]*BuildResult, prebuiltSRPMs, prebuiltDeltaSRPMs, builtSRPMs, testedSRPMs, skippedSRPMsTests, unresolvedDependencies map[string]bool, blockedSRPMs, blockedSRPMsTests map[string]*pkggraph.PkgNode, rpmConflicts, srpmConflicts []string, allowToolchainRebuilds bool, conflictsLogger func(format string, args ...interface{})) {
+	logger.Log.Info("---------------------------")
+	logger.Log.Info("--------- Summary ---------")
+	logger.Log.Info("---------------------------")
+
+	logger.Log.Infof(color.GreenString("Number of prebuilt SRPMs:           " + fmt.Sprint(len(prebuiltSRPMs))))
+	logger.Log.Infof(color.GreenString("Number of prebuilt delta SRPMs:     " + fmt.Sprint(len(prebuiltDeltaSRPMs))))
+	logger.Log.Infof(color.GreenString("Number of skipped SRPMs tests:      " + fmt.Sprint(len(skippedSRPMsTests))))
+	logger.Log.Infof(color.GreenString("Number of built SRPMs:              " + fmt.Sprint(len(builtSRPMs))))
+	logger.Log.Infof(color.GreenString("Number of tested SRPMs:             " + fmt.Sprint(len(testedSRPMs))))
+	printSummaryByVal(true, len(unresolvedDependencies), "Number of unresolved dependencies:  ")
+	printSummaryByVal(true, len(blockedSRPMs), "Number of blocked SRPMs:            ")
+	printSummaryByVal(true, len(blockedSRPMsTests), "Number of blocked SRPMs tests:      ")
+	printSummaryByVal(true, len(failedSRPMs), "Number of failed SRPMs:             ")
+	printSummaryByVal(true, len(failedSRPMsTests), "Number of failed SRPMs tests:       ")
+
+	if allowToolchainRebuilds && (len(rpmConflicts) > 0 || len(srpmConflicts) > 0) {
+		logger.Log.Infof("Toolchain RPMs conflicts are ignored since ALLOW_TOOLCHAIN_REBUILDS=y")
+	}
+
+	printSummaryByVal(!allowToolchainRebuilds, len(rpmConflicts), "Number of toolchain RPM conflicts:  ")
+	printSummaryByVal(!allowToolchainRebuilds, len(rpmConflicts), "Number of toolchain SRPM conflicts: ")
+}
+
+// Helper function to print summary in specific color, based on val and cond.
+func printSummaryByVal(cond bool, val int, msg string) {
+	if cond && (val > 0) {
+		logger.Log.Errorf(color.RedString(msg + fmt.Sprint(val)))
+	} else {
+		logger.Log.Infof(color.GreenString(msg + fmt.Sprint(val)))
+	}
+}
```

Comment:
The name of the function and the logic is getting a bit cryptic. My one suggestion to clean it up a little is to turn its signature to `(<condition>, <var_args_for_printing>)`, where:
- `<condition>` is a `bool` detrmining if we print an error or an info. For instance `len(failedSRPMs)` or `!allowToolchainRebuilds && len(rpmConflicts)`.
- `<var_args_for_printing>` is a variable number of arguments, which can be passed to `fmt.Sprintf()`.

If you're curious how to use variable argument lists, `shell.go` has many examples. You can also read more about it online ([this article](https://golangdocs.com/variadic-functions-in-golang) looks interesting for instance).

I'd also then re-name the function to something a bit more descriptive. Maybe `printErrorInfoOnCondition`? I'm not good with names here, but the goal is to give the reader a greater chance at understanding what the function will do from the name.:)

Reply:
Actually, I see that `RedString()` already accepts `<format>, <var_args>`, so you can probably skip `fmt.Sprintf`.

Reply:
Reopening. The current version with an additional `boolToInt` helper function seems like a bit of an overkill. Please re-consider my suggestion above and let me know, if you have questions or objections.